diff --git a/Makefile.am b/Makefile.am
index 735bc12..0d4e988 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 # Library.
 lib_LTLIBRARIES = libsnappy.la
 libsnappy_la_SOURCES = snappy.cc snappy-sinksource.cc snappy-stubs-internal.cc snappy-c.cc
-libsnappy_la_LDFLAGS = -version-info $(SNAPPY_LTVERSION)
+libsnappy_la_LDFLAGS = -version-info $(SNAPPY_LTVERSION) -no-undefined

 include_HEADERS = snappy.h snappy-sinksource.h snappy-stubs-public.h snappy-c.h
 noinst_HEADERS = snappy-internal.h snappy-stubs-internal.h snappy-test.h

