diff --git a/util/env_posix.cc b/util/env_posix.cc
index ad56131..7edfee8 100644
--- a/util/env_posix.cc
+++ b/util/env_posix.cc
@@ -1,6 +1,7 @@
 // Copyright (c) 2011 The LevelDB Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file. See the AUTHORS file for names of contributors.
+#if !defined(LEVELDB_PLATFORM_WINDOWS)

 #include <dirent.h>
 #include <errno.h>
@@ -603,3 +604,5 @@ Env* Env::Default() {
 }

 }  // namespace leveldb
+
+#endif
\ No newline at end of file

